src/test/java/uk/nhs/hee/tis/trainee/forms/api/ConditionsOfJoiningResourceTest.java

```diff
@@ -0,0 +1,457 @@
+              "tisId": "%s",
+              "programmeName": "Test Programme",
+              "conditionsOfJoining": {
+                "version": "%s",
+                "signedAt": "%s"
+              },
+              "signature": {
+                "signedAt": "%s",
+                "validUntil": "%s"
+              }
+            }
+            """.formatted(programmeMembershipId, version, signedAt, Instant.MIN, Instant.MAX),
+        secretKey);
+
+    String key = "40/forms/coj/" + programmeMembershipId + ".pdf";
+    when(service.getUploadedPdf(key)).thenReturn(Optional.empty());
+
+    byte[] response = "response content".getBytes();
+    Resource resource = mock(Resource.class);
+    when(resource.getContentAsByteArray()).thenReturn(response);
+    ArgumentCaptor<ConditionsOfJoiningPdfRequestDto> requestCaptor = ArgumentCaptor.captor();
+    when(service.generateConditionsOfJoining(requestCaptor.capture(), anyBoolean())).thenReturn(
+        resource);
+
+    mockMvc.perform(put("/api/coj")
+            .contentType(MediaType.APPLICATION_JSON)
+            .content(signedBody)
+            .header(HttpHeaders.AUTHORIZATION, TestJwtUtil.generateTokenForTisId("40")))
+        .andExpect(status().isOk());
+
+    ConditionsOfJoiningPdfRequestDto request = requestCaptor.getValue();
+    assertThat("Unexpected trainee ID.", request.traineeId(), is("40"));
+    assertThat("Unexpected programme membership ID.", request.programmeMembershipId(),
+        is(programmeMembershipId));
+    assertThat("Unexpected programme name.", request.programmeName(), is("Test Programme"));
+
+    ConditionsOfJoining conditionsOfJoining = request.conditionsOfJoining();
+    assertThat("Unexpected version.", conditionsOfJoining.version(), is(version));
+    assertThat("Unexpected signed at.", conditionsOfJoining.signedAt(), is(signedAt));
+  }
+
+  @ParameterizedTest
+  @EnumSource(GoldGuideVersion.class)
+  void putShouldNotPublishPdfWhenGeneratingNewPdf(GoldGuideVersion version) throws Exception {
+    UUID programmeMembershipId = UUID.randomUUID();
+
+    String signedBody = SignatureTestUtil.signData("""
+            {
+              "tisId": "%s",
+              "programmeName": "Test Programme",
+              "conditionsOfJoining": {
+                "version": "%s",
+                "signedAt": "%s"
+              },
+              "signature": {
+                "signedAt": "%s",
+                "validUntil": "%s"
+              }
+            }
+            """.formatted(programmeMembershipId, version, Instant.now(), Instant.MIN, Instant.MAX),
+        secretKey);
+
+    String key = "40/forms/coj/" + programmeMembershipId + ".pdf";
+    when(service.getUploadedPdf(key)).thenReturn(Optional.empty());
+
+    byte[] response = "response content".getBytes();
+    Resource resource = mock(Resource.class);
+    when(resource.getContentAsByteArray()).thenReturn(response);
+    when(service.generateConditionsOfJoining(any(), eq(false))).thenReturn(resource);
```

Comment:
It took me a while to figure out that the `eq(false)` is the important bit of this test 😁

Reply:
I should probably move that to a verification instead, I'll refactor it when I fix the checkbox issue 👍🏻